awx/network_ui/consumers.py

```diff
@@ -0,0 +1,277 @@
+# Copyright (c) 2017 Red Hat, Inc
+# In consumers.py
+from channels import Group, Channel
+from channels.sessions import channel_session
+from awx.network_ui.models import Topology, Device, Link, Client, Interface
+from awx.network_ui.models import TopologyInventory
+import urlparse
+from django.db.models import Q
+from collections import defaultdict
+import logging
+
+
+from awx.network_ui.utils import transform_dict
+from pprint import pformat
+
+import json
+# Connected to websocket.connect
+
+HISTORY_MESSAGE_IGNORE_TYPES = ['DeviceSelected',
+                                'DeviceUnSelected',
+                                'LinkSelected',
+                                'LinkUnSelected',
+                                'MouseEvent',
+                                'MouseWheelEvent',
+                                'KeyEvent']
+
+
+logger = logging.getLogger("awx.network_ui.consumers")
+
+
+class NetworkUIException(Exception):
+
+    pass
+
+
+def parse_inventory_id(data):
+    inventory_id = data.get('inventory_id', ['null'])
+    try:
+        inventory_id = int(inventory_id[0])
+    except ValueError:
+        inventory_id = None
+    if not inventory_id:
+        inventory_id = None
+    return inventory_id
+
+# Persistence
+
+
+class _Persistence(object):
+
+    def handle(self, message):
+        topology_id = message.get('topology')
+        assert topology_id is not None, "No topology_id"
+        client_id = message.get('client')
+        assert client_id is not None, "No client_id"
+        data = json.loads(message['text'])
+        if isinstance(data[1], list):
+            logger.error("Message has no sender")
+            return
+        if isinstance(data[1], dict) and client_id != data[1].get('sender'):
+            logger.error("client_id mismatch expected: %s actual %s", client_id, data[1].get('sender'))
+            logger.error(pformat(data))
+            return
+        message_type = data[0]
+        message_value = data[1]
+        handler = self.get_handler(message_type)
+        if handler is not None:
+            try:
+                handler(message_value, topology_id, client_id)
+            except NetworkUIException, e:
+                Group("client-%s" % client_id).send({"text": json.dumps(["Error", str(e)])})
+                raise
+            except Exception, e:
+                Group("client-%s" % client_id).send({"text": json.dumps(["Error", "Server Error"])})
+                raise
+            except BaseException, e:
```

Comment:
As of 2.5 `Exception` inherits from `BaseException` so this `except` can be removed. Also I don't see anywhere you raise a `NetworkUIException` and there is no difference in how the exception is handled. I would delete it. And up date this block to just handling `Exception` OR better, to handle the known exceptions that can come out of calling `handle`

Reply:
I will reduce this to just the except Exception clause.